crates/opencascade-sys/src/lib.rs

```diff
@@ -221,6 +221,10 @@ pub mod ffi {
         ) -> UniquePtr<BRepPrimAPI_MakeBox>;
         pub fn Shape(self: Pin<&mut BRepPrimAPI_MakeBox>) -> &TopoDS_Shape;
 
+        type BRepPrimAPI_MakeSphere;
+        pub fn BRepPrimAPI_MakeSphere_ctor(r: f64) -> UniquePtr<BRepPrimAPI_MakeSphere>;
+        pub fn Shape(self: Pin<&mut BRepPrimAPI_MakeSphere>) -> &TopoDS_Shape;
```

Comment:
Since these are all inheritance based. What do you think about creating macros to cover common sub-classes? IE to expose `IsDone, Done, Shape, etc...`

Reply:
I do think a macro is needed, but for now I've been explicitly writing everything out to observe the various patterns that fall out from the bindings.

The more cumbersome part so far has been generating constructors, and calling methods which are inherited from a base class.

I'm hoping for improvements from the `cxx` crate over time to alleviate some of these issues, but if they don't come soon then I think a macro would be a good middle ground.

Reply:
Unfortunately the cxx crate does not allow macros in the `extern "C++" ...` section. So unless we code gen it, I think its going to be manual.